src/routes/github/create-branch/github-create-branch-post.ts

```diff
@@ -0,0 +1,34 @@
+import { Request, Response } from "express";
+import { createUserClient } from "~/src/util/get-github-client-config";
+
+export const GithubCreateBranchPost = async (req: Request, res: Response): Promise<void> => {
+	const { githubToken, jiraHost } = res.locals;
+	const { owner, repo, sourceBranchName, newBranchName } = req.body;
+
+	if (!githubToken || !jiraHost) {
+		res.sendStatus(401);
+		return;
+	}
+
+	if (!owner || !repo || !sourceBranchName || !newBranchName) {
+		res.status(400).json({ err: "Missing required data." });
+		return;
+	}
+
+	try {
+		const gitHubUserClient = await createUserClient(githubToken, jiraHost, req.log, undefined);
+		const { data: baseBranchRef } = await gitHubUserClient.getReference(owner, repo, sourceBranchName);
```

Comment:
If we send the SHA for the base branch from the UI we can skip this call entirely

Reply:
The list of source branch in the frontend will need to be pulled from another API request. So, I think we will have `SHA` present for the branch and in that case we can directly pass the `SHA` value for the source branch and can skip this extra call.

Reply:
`SHA` value for source branch keeps on changing if branch updated with commit, PR and so on. It may be possible that new branch created with old SHA value if source branch updated from the time source branches pulled and hit 'create' button. I think, it is better to fetch `SHA` value just before creating branch.

Reply:
Good shout